system-tests-new/git/test_git.py

```diff
@@ -0,0 +1,90 @@
+# ##############################################################################
```

Comment:
Is most of this lifted and shifted from the old test_GrabGit?

Reply:
No, that previous test module had lots of tests targeting the specific inner workings of a much more complicated git solution. This PR greatly simplifies our git module so we wrote these simpler tests from scratch.